spanner/read.go

```diff
@@ -672,7 +676,19 @@ func (d *resumableStreamDecoder) tryRecv(mt *builtinMetricsTracer, retryer gax.R
 	if d.err == nil {
 		d.q.push(res)
 		if res.GetLast() {
-			d.changeState(finished)
+			d.changeState(ended)
+			go func(s streamingReceiver) {
+				for {
+					_, err := s.Recv()
+					if err == io.EOF {
+						break
+					}
+					if err != nil {
+						break
+					}
+				}
+
+			}(d.stream)
```

Comment:
We should also call the `cancel()` method here (assuming that we set the reference to `nil` before this, so that `iterator.Stop` does not call it). That ensures that the context is marked as done.

Reply:
I have updated the code to not set cancel nil, but checking while stopping iter does not call cancel, and is it not always the case that context is marked as done after receiving io.EOF in GRPC?